src/Symfony/Component/Security/Http/EventListener/PasswordPolicyListener.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\Credentials\PasswordCredentials;
+use Symfony\Component\Security\Http\Authenticator\Passport\Policy\PolicyInterface;
+use Symfony\Component\Security\Http\Event\CheckPassportEvent;
+
+final class PasswordPolicyListener implements EventSubscriberInterface
+{
+    /**
+     * @var array<PolicyInterface> $policies
+     */
+    private array $policies = [];
+
+    public function add(PolicyInterface $policy): void
```

Comment:
you should inject an iterable into the constructor instead, which would allow the bundle to configure that using an IteratorArgument that lazy-loads the services